clients/drcachesim/tests/drcachesim_unit_tests.cpp

```diff
@@ -313,6 +336,129 @@ LLC {
            num_accesses - 1);
 }
 
+void
+unit_test_exclusive_cache()
+{
+    // Create simple 3-level cache with exclusive LLC.
+    std::string config = R"MYCONFIG(// 3-level with exclusive LLC.
+num_cores       1
+line_size       64
+coherent        true
+
+L1I {
+  type            instruction
+  core            0
+  size            256
+  assoc           1
+  prefetcher      none
+  parent          L2
+}
+L1D {
+  type            data
+  core            0
+  size            256
+  assoc           1
+  prefetcher      none
+  parent          L2
+}
+L2 {
+  size            4K
+  assoc           4
+  inclusive       true
+  prefetcher      none
+  parent          LLC
+}
+LLC {
+  size            64K
+  assoc           4
+  exclusive       true
+  prefetcher      none
+  parent          memory
+}
+)MYCONFIG";
+    std::istringstream config_in(config);
+    cache_simulator_t cache_sim(&config_in);
+
+    // The cache config specified coherence, and the only level with
+    // multiple caches is L1.  So there should be 2 snooped caches.
+    TEST_EQ(cache_sim.get_num_snooped_caches(), 2);
+
+    // L1s are 1-way, while L2 and LLC are both 4-way.
+    // If we cycle through 4 conflicting lines multiple times, the L2 will
+    // hold all four lines and never evict anything to LLC: we expect all
+    // misses in L1, 4 misses and many hits in L2, 4 misses and no hits
+    // in LLC.
+
+    // Test 4 conflicting lines.
+    const int NUM_LOOPS = 16;
+    const int L2_ASSOC = 4;
+    const int LLC_ASSOC = 4;
+    const int LLC_SIZE = 64 * 1024;
```

Comment:
Add an assert that the parsed config has this same size.  Or just get rid of the constant and obtain the value from cache_sim.

Reply:
Either option is trickier than it sounds since there's no straightforward way to query the cache parameters from a cache_simulator_t object.  But that would improve the test, so I made a wrapper that exposes the cache_t objects so I can check the parameters.